### ISSUE TITLE
wrong output for lists

### ISSUE THREAD

**Issue by philippkuehn** [deanblackborough/php-quill-renderer#109]
Hi, we found some issues with your package with this quill delta:

```json
{
  "ops": [
    {
      "insert": "Headline 1"
    },
    {
      "attributes": {
        "header": 1
      },
      "insert": "\n"
    },
    {
      "insert": "Some text. "
    },
    {
      "attributes": {
        "bold": true
      },
      "insert": "Bold Text"
    },
    {
      "insert": ". "
    },
    {
      "attributes": {
        "italic": true
      },
      "insert": "Italic Text"
    },
    {
      "insert": ". "
    },
    {
      "attributes": {
        "italic": true,
        "bold": true
      },
      "insert": "Bold and italic Text"
    },
    {
      "insert": ". Here is a "
    },
    {
      "attributes": {
        "link": "https://scrumpy.io"
      },
      "insert": "Link"
    },
    {
      "insert": ". \nHeadline 2"
    },
    {
      "attributes": {
        "header": 2
      },
      "insert": "\n"
    },
    {
      "insert": "ordered list item"
    },
    {
      "attributes": {
        "list": "ordered"
      },
      "insert": "\n"
    },
    {
      "insert": "ordered list item"
    },
    {
      "attributes": {
        "list": "ordered"
      },
      "insert": "\n"
    },
    {
      "insert": "ordered list item"
    },
    {
      "attributes": {
        "list": "ordered"
      },
      "insert": "\n"
    },
    {
      "insert": "unordered list item"
    },
    {
      "attributes": {
        "list": "bullet"
      },
      "insert": "\n"
    },
    {
      "insert": "unordered list item with "
    },
    {
      "attributes": {
        "bold": true,
        "link": "https://scrumpy.io"
      },
      "insert": "link"
    },
    {
      "attributes": {
        "list": "bullet"
      },
      "insert": "\n"
    },
    {
      "insert": "unordered list item"
    },
    {
      "attributes": {
        "list": "bullet"
      },
      "insert": "\n"
    },
    {
      "insert": "Some Text.\n"
    }
  ]
}
```

This is the expected output:

```html
<h1>Headline 1</h1>
<p>Some text. <strong>Bold Text</strong>. <em>Italic Text</em>. <strong><em>Bold and italic Text</em></strong>. Here is a <a href="https://scrumpy.io">Link</a>. </p>
<h2>Headline 2</h2>
<ol>
<li>ordered list item</li>
<li>ordered list item</li>
<li>ordered list item</li>
</ol>
<ul>
<li>unordered list item</li>
<li>unordered list item with <a href="https://scrumpy.io"><strong>link</strong></a></li>
<li>unordered list item</li></ul>
<p>Some Text.</p>
```

This is the current output:

```html
<h1>Headline 1</h1>
<p>Some text. <strong>Bold Text</strong>. <em>Italic Text</em>. <em><strong>Bold and italic Text</strong></em>. Here is a <a href="https://scrumpy.io">Link</a>.
<br />
<h2>Headline 2</h2>
<ol>
<li>ordered list item</li>
<li>ordered list item</li>
<li>ordered list item</li>
<li>unordered list item</li>
<li>unordered list item with <a href="https://scrumpy.io">link</a><strong>link</strong></li> <li>unordered list item</li>
</ul>
Some Text.<br />
</p>
```

There are three issues here.
- there is only one wrapping paragraph around the lists
- there is an `<ol>` opening tag and a `</ul>` closing tag
- there is duplicated text output for a strong link within a list item

**Comment by deanblackborough** [comment]
Resolved in v3.17.0

**Comment by deanblackborough** [comment]
Turns out this issue isn't fixed. 

In fixing #117 I noticed that the link within the last list item is incorrect. I know what the problem is and have an idea for a fix, which I am coding now, I need to use compound inserts. 

v3.17.1 will probably be out tomorrow with the fix, at the latest the end of the week.
